typescript/web/src/components/import-button/import-images-modal/modal-dropzone/import-dropped-files/import-datasets.tsx

```diff
@@ -0,0 +1,95 @@
+import { ApolloClient, gql } from "@apollo/client";
+import { ExportFormat } from "@labelflow/graphql-types";
+import { v4 as uuidv4 } from "uuid";
+import mime from "mime-types";
+
+import Bluebird from "bluebird";
+import { uploadFile } from "../../../../../utils/upload-file";
+import { DroppedFile, SetUploadStatuses } from "../../types";
+
+import { CONCURRENCY } from "../../constants";
+
+const importDatasetMutation = gql`
+  mutation importDataset(
+    $where: DatasetWhereUniqueInput!
+    $data: DatasetImportInput!
+  ) {
+    importDataset(where: $where, data: $data) {
+      error
+    }
+  }
+`;
+const importDataset = async ({
+  apolloClient,
+  datasetId,
+  workspaceId,
+  file,
+}: {
+  apolloClient: ApolloClient<any>;
+  datasetId: string;
+  workspaceId: string;
+  file: Blob;
+}) => {
+  const key = `${workspaceId}/${datasetId}/import-datasets/${uuidv4()}-${new Date().toISOString()}.${mime.extension(
+    file.type
+  )}`;
```

Comment:
What's the purpose of this key?
Can you make is simpler by using intermediary variables `id`, `now` and `extension`?

Reply:
The purpose of this key is to upload the file to import (zip or json) into the storage. Then we can process it server side.

Reply:
I did the change on the syntax in 22e1151b6f3af9e0d8da1a51ce3784657e231242